### PR TITLE
Submit Argo workflow

### PR DESCRIPTION
**Pull request by Yancey0623**
This PR is part of #1066, add `SubmitWorkflow` implement of `Run` interface to call `codegen_couler.go`.
TODO:
1. Implement `codegen_couler.go`
1. Execute cmd to submit Argo workflow and fetch workflow ID from stdout.

### CODE REVIEW

File: cmd/sqlflowserver/main.go [hunk 1]
```diff
@@ -81,6 +85,7 @@ func main() {
 	caCrt := flag.String("ca-crt", "", "CA certificate file.")
 	caKey := flag.String("ca-key", "", "CA private key file.")
 	port := flag.Int("port", 50051, "TCP port to listen on.")
+	isArgoMode := flag.Bool("argo-mode", false, "Enable Argo workflow model.")
```
Comment (typhoonzero) [inline cmd/sqlflowserver/main.go:88]: maybe also need to configure the k8s API endpoint address in the future?

File: pkg/server/proto/sqlflow.proto [hunk 2]
```diff
@@ -10,27 +10,29 @@ service SQLFlow {
     // SQL statements like `SELECT ...`, `DESCRIBE ...` returns a rowset.
     // The rowset might be big. In such cases, Query returns a stream
     // of RunResponse
+    // 
+    // SQLFlow implements the Run interface with two mode:
     //
+    // 1. Local model
```
Comment (typhoonzero) [inline pkg/server/proto/sqlflow.proto:16]: will the "local mode" move to couler "docker mode"?

File: pkg/server/sqlflowserver.go [hunk 3]
```diff
@@ -82,6 +78,9 @@ func (s *Server) Run(req *pb.Request, stream pb.SQLFlow_RunServer) error {
 			res, err = encodeRow(s)
 		case string:
 			res, err = encodeMessage(s)
+		case sf.WorkflowJob:
+			job := r.(sf.WorkflowJob)
+			res = &pb.Response{Response: &pb.Response_Job{Job: &pb.Job{Id: job.JobID}}}
```
Comment (typhoonzero) [inline pkg/server/sqlflowserver.go:83]: If you also update the pysqlflow, please refer to this PR in the pysqlflow PR.

**Review by typhoonzero — DISMISSED** [review]
Approve with some enhancement comments

File: cmd/sqlflowserver/main.go (hunk 1 above)
Comment (Yancey0623) [reply cmd/sqlflowserver/main.go:88]: Good point and I will add it when implementing the `Fetch` interface.

File: pkg/server/proto/sqlflow.proto (hunk 2 above)
Comment (Yancey0623) [reply pkg/server/proto/sqlflow.proto:16]: The `local mode` here is the current implementation. Will move to `docker mode` if the `docker mode` is ready in Couler.

File: pkg/server/sqlflowserver.go (hunk 3 above)
Comment (Yancey0623) [reply pkg/server/sqlflowserver.go:83]: Sure, there is a task in the TODO list #1066

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**